aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -35,8 +35,12 @@ void binaryOpTensor(const Tensor& self_t, const Tensor& other_t, const Tensor& o
   Tensor self = is_self_scalar ? self_t : self_t.contiguous(at::MemoryFormat::Contiguous);
   Tensor other = is_other_scalar ? other_t : other_t.contiguous(at::MemoryFormat::Contiguous);
 
-  const MPSDataType self_dtype = getMPSScalarType((is_self_scalar && !is_other_scalar ? other_t : self_t).scalar_type());
-  const MPSDataType other_dtype = getMPSScalarType((!is_other_scalar ? other_t : self_t).scalar_type());
+  // const MPSDataType self_dtype = getMPSScalarType((is_self_scalar && !is_other_scalar ? other_t : self_t).scalar_type());
```

Comment:
I think this breaks [self:scalar, other:scalar) case. Have you tested all these cases: [self:tensor, other:tensor], [self:tensor, other:scalar], [self:scalar, other:tensor] and [self:scalar, other:scalar].

Reply:
I'll test all the cases, I made this change because, if the two inputs are of different types, and one is a scalar, then the scalar type was not being identified correctly.

Reply:
Also, according to the signature, both self_t and other_t are of type Tensor, so scalar_type() should be callable for them

Reply:
I tested all the cases, and they work (with correct data types being identified) with the change